core/node/consensus/src/tests.rs

```diff
@@ -515,3 +517,45 @@ async fn test_centralized_fetcher(from_snapshot: bool) {
     .await
     .unwrap();
 }
+
+/// Tests that generated L1 batch witnesses can be verified successfully.
+/// TODO: add tests for verification failures.
```

Comment:
will it be resolved within this PR?

Reply:
no, but will be resolved before we start using this logic. I don't have time right now to work on it.